File: patterns/trennung-von-verhaltensweisen/rules-js/trennung-von-verhaltensweisen-alarm.js
rules.JSRule({
  name: "Send message",
  triggers: [
    triggers.ItemCommandTrigger("Notification_Proxy_Info"),
    triggers.ItemCommandTrigger("Notification_Proxy_Alert")
  ],
  execute: (event) => {
    const message = String(event.receivedCommand);
    try {
      if (event.itemName === "Notification_Proxy_Info") {
        actions.NotificationAction.sendNotification("admin@example.com", message);
      } else {
        actions.NotificationAction.sendBroadcastNotification(message);
      }
      console.log(event.itemName + " -> Cloud-Benachrichtigung gesendet: " + message);
    } catch (e) {
      console.warn("Cloud-Benachrichtigung nicht verfuegbar (" + event.itemName + "): " + e.message + " - Nachricht: " + message);
    }
  }
});

File: patterns/trennung-von-verhaltensweisen/rules-python/trennung_von_verhaltensweisen_alarm.py
from openhab import rule, logger
from openhab.actions import NotificationAction
from openhab.triggers import ItemCommandTrigger


@rule(triggers=[
    ItemCommandTrigger("Notification_Proxy_Info"),
    ItemCommandTrigger("Notification_Proxy_Alert"),
])
class SendMessage:
    def execute(self, module, input):
        event = input.get("event")
        if not event:
            return

        item_name = event.getItemName()
        command = str(event.getItemCommand())

        try:
            if item_name == "Notification_Proxy_Info":
                if NotificationAction is not None:
                    NotificationAction.sendNotification("admin@example.com", command)
                logger.info("Info-Benachrichtigung: " + command)
            else:
                if NotificationAction is not None:
                    NotificationAction.sendBroadcastNotification(command)
                logger.info("Alarm-Benachrichtigung: " + command)
        except Exception as ex:
            logger.warn("Cloud-Benachrichtigung nicht verfuegbar: " + str(ex))
